### DISCUSSION TITLE
Rust-based replacement for OSRM

### DISCUSSION THREAD

**Discussion by do-me** [reinterpretcat/vrp#155] [General]
Hey, 

I was wondering whether anyone knows any good OSRM drop-in replacements that are faster (and possibly require less RAM), especially for large areas like all of USA or all of EU. If possible based on the pretty reliable Geofabrik subsets.

For now, I found two Rust-based candidates: 

1. [Fast Paths](https://github.com/easbar/fast_paths)

Seems to be indeed fast according to the stats in the readme (all of US contracted in 5-10 mins). It can run in-browser and is used e.g. in https://github.com/Urban-Analytics-Technology-Platform/od2net for dynamically retrieving OSM data via Overpass and contracting on-the-fly.

Would probably make for a nice browser-based demo for VRP too, having all the components running in the browser.

2. [Ptolemy](https://github.com/sitegui/ptolemy)

Very fast too (Brazil in <1min) and partially compatible with OSRM API endpoints.

P.S. @reinterpretcat: It's great you're back! Your work rocks, VRP is by far the most feature-complete solver out there :)

**Comment by reinterpretcat** [comment]
thanks for feedback!

btw, I've added some examples how to extend the solver (this was one of the major changes): 
https://github.com/reinterpretcat/vrp/blob/master/vrp-core/examples/custom_constraint.rs
https://github.com/reinterpretcat/vrp/blob/master/vrp-core/examples/custom_objective.rs

**Comment by do-me** [comment]
Great news: OSRM team is thinking about re-implementing OSRM in Rust!
https://x.com/ProjectOSRM/status/1825219772050456796
